Google.PowerShell/PubSub/GcpsCmdlet.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Google.Apis.Pubsub.v1;
+using Google.PowerShell.Common;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.PowerShell.PubSub
+{
+    /// <summary>
+    /// Base class for Google Cloud PubSub Cmdlets.
+    /// </summary>
+    public class GcpsCmdlet : GCloudCmdlet
+    {
+        public PubsubService Service { get; private set; }
+
+        public GcpsCmdlet()
+        {
+            Service = new PubsubService(GetBaseClientServiceInitializer());
+        }
+
+        /// <summary>
+        /// Prefix projects/{project name}/topics/{topic} to topicName if not present.
+        /// </summary>
+        protected string PrefixProjectToTopic(string topicName, string project)
+        {
+            if (string.Equals(topicName, "_deleted-topic_", System.StringComparison.OrdinalIgnoreCase))
```

Comment:
Upon further testing, it seems "_deleted-topic_" is not recognized by the PubSub API at all so I removed it.